### ISSUE TITLE
🐛 [Bug] aliased_io is lost on an engine cache hit, changing forward() output arity

### ISSUE THREAD

**Issue by shoumikhin** [pytorch/TensorRT#4521]
## Bug Description

`aliased_io` is not round-tripped through the engine cache. A fresh build records it, but a
cache hit reconstructs `SerializedInterpreterResult` without it, so it silently falls back
to the `{}` default.

The visible effect is that **the same model returns a different number of outputs depending
on whether the engine came from cache**. For an aliased/KV-cache engine, the interpreter
appends a side-effect output to satisfy the network-output requirement of
`IKVCacheUpdateLayer`. `TorchTensorRTModule.forward` truncates that output using
`aliased_io`:

```python
# py/torch_tensorrt/dynamo/runtime/_TorchTensorRTModule.py
if self.aliased_io:
    n = user_output_count(self.output_binding_names, self.aliased_io)
    if n < len(outputs):
        outputs = outputs[:n]
```

With an empty map the guard is skipped entirely, no truncation happens, and the internal
side-effect tensor leaks into the caller's return value.

Using `user_output_count` from main unmodified, with `output_binding_names = ["output0",
"output1"]` where `output1` is the appended side-effect output:

```
FRESH build  aliased_io={'output1': ('buf_cache', 'kv_cache_update')}
   user_output_count = 1  -> forward() returns 1 tensor

CACHE HIT    aliased_io={}
   user_output_count = 2  -> forward() returns 2 tensors
```

Same engine, different Python-visible output arity. Code that does `logits = model(x)` gets
a tuple instead of a tensor, or unpacking fails, only on the second run.

## Where it is lost

`py/torch_tensorrt/dynamo/conversion/_conversion.py`

The cache insert tuple omits it (7 fields, no `aliased_io`):

```python
engine_cache.insert(
    hash_val,
    (
        serialized_engine_in_cache,
        interpreter_result.input_names,
        interpreter_result.output_names,
        inputs,
        settings,
        interpreter_result.requires_output_allocator,
        interpreter_result.requires_native_multidevice,
    ),
)
```

And `pull_cached_engine` rebuilds the result without it, so the `{}` default applies:

```python
return SerializedInterpreterResult(
    serialized_engine=serialized_engine,
    input_names=input_names,
    output_names=output_names,
    requires_output_allocator=requires_output_allocator,
    requires_native_multidevice=requires_native_multidevice,
    symbolic_shape_expressions=symbolic_shape_expressions,
    # aliased_io missing -> defaults to {}
)
```

The fresh path does set it (`aliased_io=interpreter_result.aliased_io`), which is the
asymmetry.

This looks like an oversight in #4251, which introduced `aliased_io` and wired the fresh
build path and the `TorchTensorRTModule` consumer, but not the cache insert/pull. No later
commit to `_conversion.py` addresses it.

## Scope, and what is NOT affected

Engine execution itself is fine. Both runtimes reconcile against
`ICudaEngine::getAliasedInputTensor`, which is the source of truth for TRT-enforced
aliasing, and recover a `kv_cache_update` alias the serialized map lacks:

- `py/torch_tensorrt/dynamo/runtime/_TRTEngine.py` `_reconcile_aliased_io`
- `core/runtime/TRTEngine.cpp` (same logic)

So the in-place write still happens. What is *not* recovered is the Python wrapper's own
`self.aliased_io`, which is set only from the constructor argument and is what
`forward`'s truncation and `_pack_engine_info`'s `serialize_aliased_io` read.

Two consequences follow:

1. **Output arity differs on a cache hit** (shown above).
2. **`kind="user"` aliases would be lost outright.** TRT does not track user-declared
   aliases, so reconciliation cannot recover them by design. On current main only
   `AliasKind.KV_CACHE_UPDATE` is ever produced (`conversion/impl/slice_scatter.py`), so
   this part is latent rather than active, but it makes the cache path unsafe for the
   user-declared aliasing the type already supports.

## Expected behavior

A cache hit should produce the same `SerializedInterpreterResult` as a fresh build, so
`aliased_io` should be stored in the cached tuple and restored on retrieval. Note this
changes the cached tuple layout, so previously cached entries would unpack with the wrong
arity. I did not find an existing cache-version constant in `_engine_cache.py`, so some
form of backward compatibility (a length check on `cached_data`, or invalidating old
entries) probably needs deciding as part of the fix.

## Environment

- Torch-TensorRT: main at `dd0afdb7e`
- Verified by reading `origin/main` and executing main's own `user_output_count`

## Additional context

Found while reviewing #4470, which adds a `lift_mutated_buffers` prediction cross-check
that reads the wrapper's `aliased_io`. That cross-check is not on main, so this report is
deliberately scoped to what main itself does. Filing separately because the cache gap is
independent of that PR and predates it.

I have not reproduced this on a GPU with a real cache hit (no GPU/TensorRT in my
environment); the arity result above comes from running main's `user_output_count` directly
against the two map states, plus tracing the insert/pull paths. Happy to send a PR.
